pkg/generator/changelog_generator.go

```diff
@@ -36,18 +37,61 @@ var typeToText = map[string]string{
 	"refactor": "Code Refactoring",
 	"test":     "Tests",
 	"chore":    "Chores",
-	"%%bc%%":   "Breaking Changes",
+	"build":    "Build",
+	"ci":       "CI",
+}
+
+var typeOrder = map[int]string{
+	0:  "%%bc%%",
+	1:  "feat",
+	2:  "fix",
+	3:  "perf",
+	4:  "revert",
+	5:  "docs",
+	6:  "style",
+	7:  "refactor",
+	8:  "test",
+	9:  "chore",
+	10: "build",
+	11: "ci",
 }
 
 func getSortedKeys(m *map[string]string) []string {
-	keys := make([]string, len(*m))
+	var keys []int
+	types := make(map[int]string, len(*m))
+	sortedKeys := make([]string, len(*m))
+
 	i := 0
+	for range typeOrder {
+		keys = append(keys, i)
+		i++
+	}
+
+	sort.Ints(keys)
+
+	i = 0
 	for k := range *m {
-		keys[i] = k
+		types[i] = k
+		i++
+	}
+
+	i = 0
+	for ki := range keys {
+		for ti := range types {
+			if types[ti] == typeOrder[ki] {
+				sortedKeys[i] = types[ti]
+				delete(types, ti)
+				i++
+			}
+		}
+	}
+
+	for ti := range types {
+		sortedKeys[i] = types[ti]
 		i++
 	}
-	sort.Strings(keys)
-	return keys
+
+	return sortedKeys
 }
```

Comment:
Not completely happy with this implementation, but at least it works the way I intended it to work.